tools/c7n_azure/c7n_azure/utils.py

```diff
@@ -99,6 +100,41 @@ def azure_name_value_pair(name, value):
     return NameValuePair(**{'name': name, 'value': value})
 
 
+send_logger = logging.getLogger('custodian.azure.utils.ServiceClient.send')
+
+
+def custodian_azure_send_override(self, request, headers=None, content=None, **kwargs):
+    """ Overrides ServiceClient.send() function to implement retries & log headers
+    """
+    retries = 0
+    max_retries = 3
+    while retries < max_retries:
+        response = self.old_send(request, headers, content, **kwargs)
+
+        send_logger.debug(response.status_code)
+        for k, v in response.headers.items():
+            if k.startswith('x-ms-ratelimit'):
+                send_logger.debug(k + ':' + v)
+
+        # Retry codes from urllib3/util/retry.py
+        if response.status_code in [413, 429, 503]:
```

Comment:
Do you think this merits a warning level log? Just thinking about how we can give ourselves visibility into when this is happening to help plan future enhancements here.